src/main/java/com/jvm_bloggers/core/rss/json/RssToJsonConverter.java

```diff
@@ -0,0 +1,100 @@
+package com.jvm_bloggers.core.rss.json;
+
+import com.jvm_bloggers.core.rss.BlogPostsController;
+import com.rometools.rome.feed.synd.SyndEntry;
+import com.rometools.rome.feed.synd.SyndFeed;
+
+import javaslang.control.Option;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.wicket.ajax.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static com.jvm_bloggers.core.rss.json.RssJsonKey.AUTHOR;
+import static com.jvm_bloggers.core.rss.json.RssJsonKey.DATE;
+import static com.jvm_bloggers.core.rss.json.RssJsonKey.DESCRIPTION;
+import static com.jvm_bloggers.core.rss.json.RssJsonKey.ENTRIES;
+import static com.jvm_bloggers.core.rss.json.RssJsonKey.GENERATOR;
+import static com.jvm_bloggers.core.rss.json.RssJsonKey.LINK;
+import static com.jvm_bloggers.core.rss.json.RssJsonKey.TITLE;
+import static com.jvm_bloggers.utils.NowProvider.DEFAULT_ZONE;
+
+/**
+ * Simple {@link SyndFeed} objects to JSON plain text converter
+ *
+ * @author kraluk
+ */
+@Component
+@Slf4j
+public class RssToJsonConverter {
+    private static final DateTimeFormatter DATE_FORMATTER =
+        DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    private final String baseUrl;
+
+    @Autowired
+    public RssToJsonConverter(@Value("${application.baseUrl}") String baseUrl) {
+        this.baseUrl = baseUrl;
+    }
+
+    private static JSONObject toJson(SyndFeed feed) {
+        JSONObject json = new JSONObject();
+
+        json.put(TITLE.getKey(), feed.getTitle());
+        json.put(LINK.getKey(), feed.getLink());
+        json.put(DESCRIPTION.getKey(), feed.getDescription());
+        json.put(ENTRIES.getKey(), toJson(feed.getEntries()));
+
+        return json;
+    }
+
+    private static java.util.List<JSONObject> toJson(java.util.List<SyndEntry> entries) {
+        return javaslang.collection.List.ofAll(entries)
```

Comment:
Yep, frankly speaking, I'm not sure now, what I've wanted to do here... It was over-engineered completely. At the end, staying with java.util.List (as external libraries required it) I hope it'll be a good idea here, as you suggested :)